### PR TITLE
[7.x] Waiting before retrying password reset

### PR DESCRIPTION
**Pull request by chmv**
https://github.com/laravel/framework/pull/30340 finalization.
Method `recentlyCreatedToken` has been added to `TokenRepositoryInterface` interface and tests have been adapted.

### CODE REVIEW

File: src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php [hunk 1]
```diff
@@ -23,6 +23,14 @@ public function create(CanResetPasswordContract $user);
      */
     public function exists(CanResetPasswordContract $user, $token);
 
+    /**
+     * Determine if the given user recently created a password reset token.
+     *
+     * @param  \Illuminate\Contracts\Auth\CanResetPassword  $user
+     * @return bool
+     */
+    public function recentlyCreatedToken(CanResetPasswordContract $user);
```
Comment (gocanto) [inline src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: would it not be a breaking change for those implementing this interface?

Comment (devcircus) [reply src/Illuminate/Auth/Passwords/TokenRepositoryInterface.php]: If it's going to master and master represents the next major version, it should be ok.

**Comment by netpok** [comment]
@chmv You forgot the `?? 0` part in the provider

**Comment by chmv** [comment]
@netpok You are the best!

**Comment by netpok** [comment]
@taylorotwell 

The default zero was added in 6.x to prevent this being a breaking change. It was removed in this pull request because:
- Existing users should decide if they want this feature or not, they also should be notified of this change in the upgrade guide.
- More importantly leaving this line as is makes this feature silently getting disabled when one accidentally deletes this configuration entry.

I could make a PR in the docs for documenting this change in the upgrade.md

**Pull request merged**